nixos/modules/services/web-apps/stirling-pdf.nix

```diff
@@ -0,0 +1,179 @@
+{
+  config,
+  lib,
+  options,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    getExe
+    literalExpression
+    mkEnableOption
+    mkIf
+    mkMerge
+    mkOption
+    mkPackageOption
+    optional
+    ;
+
+  inherit (lib.types)
+    attrsOf
+    either
+    nullOr
+    path
+    port
+    str
+    ;
+
+  cfg = config.services.stirling-pdf;
+in
+
+{
+  meta.maintainers = with lib.maintainers; [ thubrecht ];
+
+  options.services.stirling-pdf = {
+    enable = mkEnableOption "Stirling-PDF, a web application for manipulating PDF files";
+
+    package = mkPackageOption pkgs "stirling-pdf" { };
+
+    port = mkOption {
+      type = port;
+      default = 8080;
+      description = ''
+        Port that Stirling-PDF will listen on.
+      '';
+    };
+
+    environment = mkOption {
+      type = attrsOf (nullOr (either str path));
+      defaultText = ''
+        {
+          SERVER_PORT = cfg.port;
+        }
+      '';
+      description = ''
+        Settings for the stirling-pdf app.
+      '';
+    };
+
+    environmentFile = mkOption {
+      type = nullOr path;
+      default = null;
+      description = ''
+        File containing additional environment variables to pass to Stirling PDF.
+      '';
+    };
+
+    domain = mkOption {
+      type = nullOr str;
+      default = null;
+      description = ''
+        The domain for the application.
+        This is used to set the Nginx options to the correct vhost.
+      '';
+    };
+
+    nginx = mkOption {
+      type = nullOr options.services.nginx.virtualHosts.type.nestedTypes.elemType;
+      default = null;
+      description = ''
+        With this option, you can customize the nginx virtualHost settings.
+      '';
+      example = literalExpression ''
+        {
+          # To enable encryption and let Let's Encrypt take care of certificate
+          forceSSL = true;
+          enableACME = true;
+        }
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.nginx != { } -> cfg.domain != null;
+        message = "A domain is required to setup nginx.";
```

Comment:
the message here isn't technically correct, since a virtualhost's name may also be set inside its config via [services.nginx.virtualHosts.<name>.serverName](https://search.nixos.org/options?channel=23.11&show=services.nginx.virtualHosts.%3Cname%3E.serverName&from=0&size=50&sort=relevance&type=packages&query=services.nginx.virtualHosts.%3Cname%3E.serverName), it just defaults to the block's attribute name.

Reply:
As we have `virtualHosts.${cfg.domain}` in our case it is required to set the `domain` option